### ISSUE TITLE
Unable to resolve URIs of combined (include/import) internal stylesheets with XSLTServlet

### ISSUE THREAD

**Issue by ptrckb** [eXist-db/exist#632]
I have the case of an XSLT stylesheet combining with another one from another collection. The URI of the second stylesheet is correctly resolved.

However if the second stylesheet also has imports, it fails to resolve their URIs unless the hrefs are relative to the first stylesheet.

For example, with this route in an application controller:

``` xquery
else if ($exist:path eq "/test") then
    <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
        <forward url="{ $exist:controller}/data.xml"/>
        <view>
            <forward servlet="XSLTServlet">
                <set-attribute name="xslt.stylesheet" value="{ $exist:root }{ $exist:controller }/xsl/test.xsl"/>
            </forward>
        </view>
    </dispatch>
```

and a toplevel stylesheet (xsl/test.xsl):

``` xml
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
    <xsl:import href="dir/stylesheet1.xsl"/>
    <xsl:template match="/">
        <x>
            <xsl:call-template name="test"/>
        </x>
    </xsl:template>
</xsl:stylesheet>
```

and lower-level stylesheets (xsl/dir/stylesheet1.xsl):

``` xml
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
    <xsl:import href="stylesheet2.xsl"/>
</xsl:stylesheet>
```

and (xsl/dir/stylesheet2.xsl):

``` xml
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
    <xsl:template name="test">
        <y>OK</y>
    </xsl:template>
</xsl:stylesheet>
```

it fails with the messages:

> Error at /xsl:stylesheet/xsl:import[1]
>   XTSE0165: I/O error reported by XML parser processing
>   file:/----------------------------/exist/imported.xsl:
>   /---------------------------/exist/imported.xsl (Aucun fichier ou dossier de ce type)
> Error at /xsl:stylesheet/xsl:import[1]
>   XTSE0165: Reported 1 error in imported stylesheet module

With the following changes to the database resolver for XSLTServlet (eXist-db 2.2), it now successfully resolves the URIs and produces the expected result:

```
diff --git a/src/org/exist/http/servlets/XSLTServlet.java b/src/org/exist/http/servlets/XSLTServlet.java
index f406c23..4ea0375 100644
--- a/src/org/exist/http/servlets/XSLTServlet.java
+++ b/src/org/exist/http/servlets/XSLTServlet.java
@@ -592,6 +592,8 @@ public class XSLTServlet extends HttpServlet {
             //TODO : use dedicated function in XmldbURI
             if(href.startsWith("/")){
                 path = href;
+            } else if(base != null && !base.isEmpty()){
+                path = XmldbURI.create(base).removeLastSegment().append(href).toString();
             } else {
                 path = collection.getURI() + "/" + href;
             }
@@ -612,7 +614,7 @@ public class XSLTServlet extends HttpServlet {
                 throw new TransformerException("Insufficient privileges to read resource " + path);
             }

-            final DOMSource source = new DOMSource(xslDoc);
+            final DOMSource source = new DOMSource(xslDoc, xslDoc.getBaseURI());
             return source;
         }
     }
```

**Comment by dizzzz** [comment]
@adamretter @shabanovd something for you?

**Comment by ptrckb** [comment]
It seems this problem has been fixed by 04c1771588dcae48489b4705099210a83ec6840c. Thanks @shabanovd
